src/coreclr/tools/aot/ILCompiler.Build.Tasks/ComputeManagedAssembliesToCompileToNative.cs

```diff
@@ -153,8 +158,21 @@ public override bool Execute()
 
                 // Remove any assemblies whose implementation we want to come from NativeAOT's package.
                 // Currently that's System.Private.* SDK assemblies and a bunch of framework assemblies.
-                if (nativeAotFrameworkAssembliesToUse.Contains(assemblyFileName))
+                if (nativeAotFrameworkAssembliesToUse.TryGetValue(assemblyFileName, out ITaskItem frameworkItem))
                 {
+                    if (GetFileVersion(itemSpec).CompareTo(GetFileVersion(frameworkItem.ItemSpec)) > 0)
```

Comment:
Conflict resolution also checks assembly version first, but I don't think our current packages have any case where we might ship an older assembly version with newer file version (EG: newer package from lower servicing band).  We used to - which is why that check in conflict resolution was required - but a single file version check should be good enough now since I think we give every assembly the same major.minor file and assembly version.   Does that sound right @ViktorHofer?

Reply:
The only exception that I know of in dotnet/runtime is `System.Runtime.Serialization.Formatters`

OOB 9.0.0 package assembly:

![image](https://github.com/user-attachments/assets/ccc1945e-9443-4bd1-8d1e-b3bfa8fdf769)

inbox 9.0.0 assembly:

![image](https://github.com/user-attachments/assets/f8311dd0-24be-4801-9385-990c08bc4dec)

There might be other packages, not sure. The above don't probably doesn't matter because only the inbox assembly has the difference, not the packaged-up assembly.